rpc/subscriptions.go

```diff
@@ -68,23 +126,54 @@ func (h *Handler) SubscribeEvents(ctx context.Context, fromAddr *felt.Felt, keys
 
 	headerSub := h.newHeads.Subscribe()
 	reorgSub := h.reorgs.Subscribe() // as per the spec, reorgs are also sent in the events subscription
+	pendingSub := h.pendingBlock.Subscribe()
 	sub.wg.Go(func() {
 		defer func() {
 			h.unsubscribe(sub, id)
 			headerSub.Unsubscribe()
 			reorgSub.Unsubscribe()
+			pendingSub.Unsubscribe()
 		}()
 
-		// The specification doesn't enforce ordering of events therefore events from new blocks can be sent before
+		// The specification doesn't enforce ordering of events, therefore, events from new blocks can be sent before
 		// old blocks.
 		var wg conc.WaitGroup
 		wg.Go(func() {
+			// Stores the transaction hash -> number of events
+			eventsPreviouslySent := make(map[blockchain.FilteredEvent]struct{})
+
 			for {
 				select {
 				case <-subscriptionCtx.Done():
 					return
 				case header := <-headerSub.Recv():
-					h.processEvents(subscriptionCtx, w, id, header.Number, header.Number, fromAddr, keys)
```

Comment:
It seems that the entire `case header` code is not being tested. Could you please add a test that follows this execution path.